include/bitcoin/bitcoin/chain/chain_state.hpp

```diff
@@ -174,6 +174,13 @@ class BC_API chain_state
     /// This block height is less than or equal to that of the top checkpoint.
     bool is_under_checkpoint() const;
 
+    static uint32_t minimum_timespan(uint32_t retargeting_interval_seconds,
+        uint32_t retargeting_factor);
+    static uint32_t maximum_timespan(uint32_t retargeting_interval_seconds,
+        uint32_t retargeting_factor);
+    static uint32_t retargeting_interval(uint32_t retargeting_interval_seconds,
+        uint32_t block_spacing_seconds);
+
```

Comment:
Style: list static functions at the top.

Reply:
Done.